daemon/api_notices.go

```diff
@@ -218,6 +230,53 @@ func allowedNoticeTypesForInterface(iface string) []state.NoticeType {
 	return types
 }
 
+func postNotices(c *Command, r *http.Request, user *auth.UserState) Response {
+	requestUID, err := uidFromRequest(r)
+	if err != nil {
+		return Forbidden("cannot determine UID of request, so cannot create notice")
+	}
+
+	var payload struct {
+		Action string `json:"action"`
+		Type   string `json:"type"`
+		Key    string `json:"key"`
+		// NOTE: Data and RepeatAfter fields are not needed for snap-run-inhibit notices.
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode request body: %v", err)
+	}
+
+	if payload.Action != "add" {
+		return BadRequest("invalid action %q", payload.Action)
+	}
+	if payload.Type != "snap-run-inhibit" {
```

Comment:
Actually, we were going to use `custom` notices for this, but the fact we had to stuff the notice type (snap run is inhibited) and key (inhibited instance name) in the `custom` notice key (e.g. `key="snap-run-inhibited/firefox"`) meant that a listener (i.e. `snapd-desktop-integration`) won't have a way to filter for `snap-run-inhibited` events since we don't have pattern based matching for notices (yet). Also, we wanted to be less flexible with custom notices since it doesn't make sense for snapd, unlike pebble. 

Basically, it is custom notices, but with a type field.